### PR TITLE
Cython optimization

### PR DESCRIPTION
**Pull request by tsedan**
Draft pr to keep the branch up to date as I incorporate the optimizations.

### CODE REVIEW

**Comment by tinmarr** [comment]
@tsedan plz fix the demos and update the changelog

File: demo/platformer.py [hunk 1]
```diff
@@ -200,7 +200,7 @@ def update():
 
     if rb.Input.key_pressed("r"):
         player.pos = rb.Display.center_left + rb.Vector(50, 0)
-        player.get(rb.RigidBody).velocity = rb.Vector.zero
+        player.get(rb.RigidBody).velocity = rb.Vector.zero()
```
Comment (tinmarr) [inline demo/platformer.py:203]: This line change needs to be reflected in the platformer tutorial.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,11 +2,17 @@
 
 ## [Unreleased]
 
+### Key Features
+-   Made `Vector` a C class
+
 ### Breaking Changes
+-   As `Vector` is now a C class, it only holds floats and is therefore subject to floating point errors in unexpected cases.
+    Be careful in accuracy-dependent calculations to handle deviation properly.
```
Comment (tinmarr) [inline CHANGELOG.md:11]:
```suggestion
-   Rename `Vector.to_int()` to `Vector.rounded()`. It now rounds a vector instead of doing an int cast

```

**Review by tinmarr — CHANGES_REQUESTED** [review]

File: demo/platformer.py (hunk 1 above)
Comment (tsedan) [reply demo/platformer.py:203]: It already is

**Review by tinmarr — APPROVED** [review]

**Pull request merged**